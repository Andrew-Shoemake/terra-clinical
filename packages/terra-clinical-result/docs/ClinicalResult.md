# Terra Clinical Result

The Clinical Result is a starndardized view for single result values added to a Patient's Medical Chart. Applies to Vital Signs, Laboratory test results, etc.


## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-clinical-result`


<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
<!-- AUTO-GENERATED-CONTENT:END -->


## Component Features

 * [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
 * [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
 * [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
 * [LTR/RTL Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#ltr--rtl-support)
