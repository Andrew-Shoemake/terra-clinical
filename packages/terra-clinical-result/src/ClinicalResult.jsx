import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import IconModified from 'terra-icon/lib/icon/IconModified';
import IconComment from 'terra-icon/lib/icon/IconComment';
import IconUnverified from 'terra-icon/lib/icon/IconDiamond';
import ResultError from './common/other/_ResultError';
import NoData from './common/other/_KnownNoData';
import Observation from './common/observation/_Observation';
import observationPropShape from './proptypes/observationPropTypes';
import styles from './ClinicalResult.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Result Object with the clinical result data.
   */
  resultData: observationPropShape,
  /**
   * Visually hides the unit of measure when presented in a series of side-by-side columns of the same unit.
   */
  hideUnit: PropTypes.bool,
  /**
   * Whether or not the text should be truncated in display. Restricts clinical result details each to one line.
   */
  isTruncated: PropTypes.bool,
  /**
   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
   */
  hasResultError: PropTypes.bool,
  /**
   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
   */
  hasResultNoData: PropTypes.bool,
};

const defaultProps = {
  resultData: {},
  hideUnit: false,
  isTruncated: false,
  hasResultError: false,
  hasResultNoData: false,
};

const createIcons = (resultData) => (resultData.isUnverified
  ? (<IconUnverified className={cx('icon-unverified')} />)
  : (
    <React.Fragment>
      {resultData.isModified ? (<IconModified className={cx('icon-modified')} />) : null}
      {resultData.hasComment ? (<IconComment className={cx('icon-comment')} />) : null}
    </React.Fragment>
  )
);

const createSecondaryDisplays = (resultData) => (
  <React.Fragment>
    {resultData.conceptDisplay ? (<div className={cx('concept-display')}>{resultData.conceptDisplay}</div>) : null}
    {resultData.datetimeDisplay ? (<div className={cx('datetime-display')}>{resultData.datetimeDisplay}</div>) : null}
  </React.Fragment>
);

const ClinicalResult = (props) => {
  const {
    resultData,
    hideUnit,
    isTruncated,
    hasResultError,
    hasResultNoData,
    ...customProps
  } = props;

  const clinicalResultDisplay = () => {
    if (hasResultError) {
      return <ResultError />;
    }
    if (hasResultNoData) {
      return <NoData />;
    }
    const decoratedResultClassnames = cx([
      'decorated-result-display',
      { truncated: isTruncated },
    ]);
    return (
      <React.Fragment>
        <div className={decoratedResultClassnames}>
          <div className={cx('result-display')}>
            <Observation
              eventId={resultData.eventId}
              result={resultData.result}
              interpretation={resultData.interpretation}
              isUnverified={resultData.isUnverified}
              hideUnit={hideUnit}
            />
            {isTruncated ? null : createIcons(resultData)}
          </div>
          {isTruncated ? createIcons(resultData) : null}
        </div>
        {createSecondaryDisplays(resultData)}
      </React.Fragment>
    );
  };

  const clinicalResultClassnames = cx([
    'clinical-result',
    customProps.className,
  ]);

  return (
    <div {...customProps} className={clinicalResultClassnames}>
      {clinicalResultDisplay}
    </div>
  );
};

ClinicalResult.propTypes = propTypes;
ClinicalResult.defaultProps = defaultProps;

export default ClinicalResult;
