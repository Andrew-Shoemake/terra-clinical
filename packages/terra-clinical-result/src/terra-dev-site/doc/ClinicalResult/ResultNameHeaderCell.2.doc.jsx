import React from 'react';
import DocTemplate from 'terra-doc-template';
import ReadMe from '../../../../docs/ResultNameHeaderCell.md';
import { name } from '../../../../package.json';

// Component Source
import ResultSrc from '!raw-loader!../../../../src/resultnameheadercell/ResultNameHeaderCell';

// Example Files
import Default from '../examples/ResultNameHeaderCell';
import DefaultSrc from '!raw-loader!../../../../src/terra-dev-site/doc/examples/ResultNameHeaderCell.jsx';

const DocPage = () => (
  <DocTemplate
    packageName={name}
    readme={ReadMe}
    srcPath={`https://github.com/cerner/terra-clinical/tree/master/packages/${name}`}
    examples={[
      {
        title: 'Result Name Header Cell (for flowsheets)',
        example: <Default />,
        source: DefaultSrc,
      },
    ]}
    propsTables={[
      {
        componentName: 'Result Name Header Cell',
        componentSrc: ResultSrc,
      },
    ]}
  />
);

export default DocPage;
