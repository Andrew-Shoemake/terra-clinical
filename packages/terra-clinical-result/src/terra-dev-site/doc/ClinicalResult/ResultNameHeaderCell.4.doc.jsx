import React from 'react';
import DocTemplate from 'terra-doc-template';
import ReadMe from '../../../../docs/ResultNameHeaderCell.md';
import { name } from '../../../../package.json';

// Component Source
import ResultSrc from '!raw-loader!../../../../src/resultnameheadercell/ResultNameHeaderCell';

// Example Files
import Default from '../example/resultnameheadercell/ResultNameHeaderCell';
import DefaultSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/resultnameheadercell/ResultNameHeaderCell.jsx';
import FlowsheetExample from '../example/resultnameheadercell/ResultNameHeaderFlowsheetExample';
import FlowsheetExampleSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/resultnameheadercell/ResultNameHeaderFlowsheetExample.jsx';

const DocPage = () => (
  <DocTemplate
    packageName={name}
    readme={ReadMe}
    srcPath={`https://github.com/cerner/terra-clinical/tree/master/packages/${name}`}
    examples={[
      {
        title: 'Result Name Header Cell',
        example: <Default />,
        source: DefaultSrc,
      },
      {
        title: 'Result Name Header Cells in Sample Flowsheet',
        example: <FlowsheetExample />,
        source: FlowsheetExampleSrc,
      },
    ]}
    propsTables={[
      {
        componentName: 'Result Name Header Cell',
        componentSrc: ResultSrc,
      },
    ]}
  />
);

export default DocPage;
