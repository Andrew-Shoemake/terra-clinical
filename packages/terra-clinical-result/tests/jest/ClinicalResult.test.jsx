import React from 'react';
import ClinicalResult, { ResultNameHeaderCell, ResultTimeHeaderCell, ClinicalResultBloodPressure } from '../../lib/index';
import DefaultResult, { DefaultBloodPressureResult, DefaultSystolicResult, DefaultDiastolicResult } from '../../src/terra-dev-site/test/clinical-result/TestResults';

// Snapshot Tests
it('should render a default component', () => {
  const nameHeaderCell = shallow(<ResultNameHeaderCell resultName="Blood Pressure" unit="mmHg" />);
  expect(nameHeaderCell).toMatchSnapshot();
});

it('should render a default ResultTimeHeaderCell component', () => {
  const timeHeaderCell = shallow(<ResultTimeHeaderCell date="Dec 21, 2010" time="23:59" />);
  expect(timeHeaderCell).toMatchSnapshot();
});

describe('ClinicalResult', () => {
  it('should render a default ClinicalResult', () => {
    const result = shallow(<ClinicalResult resultData={DefaultResult} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResult', () => {
    const result = shallow(<ClinicalResult resultData={DefaultResult} isTruncated />);
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResult with icons moved', () => {
    const resultData = {
      ...DefaultResult,
      isModified: true,
    };
    const result = shallow(<ClinicalResult resultData={resultData} isTruncated />);
    expect(result).toMatchSnapshot();
  });

  it('should pass certain props to Observation ', () => {
    const resultData = {
      ...DefaultResult,
      interpretation: 'CRITICAL',
      type: 'ALPHA',
      status: 'registered',
      performedDateTime: new Date('December 17, 1995 03:24:00'),
      updateDateTime: new Date('October 21, 1995 04:24:00'),
    };
    const result = shallow(<ClinicalResult resultData={resultData} hideUnit />);
    expect(result).toMatchSnapshot();
  });

  it('should render a modified result ', () => {
    const resultData = {
      ...DefaultResult,
      isModified: true,
    };
    const result = shallow(<ClinicalResult resultData={resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a commented result ', () => {
    const resultData = {
      ...DefaultResult,
      hasComment: true,
    };
    const result = shallow(<ClinicalResult resultData={resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render an unverified result ', () => {
    const resultData = {
      ...DefaultResult,
      isUnverified: true,
    };
    const result = shallow(<ClinicalResult resultData={resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a result with unit hidden', () => {
    const resultData = {
      ...DefaultResult,
      hideUnit: true,
    };
    const result = shallow(<ClinicalResult resultData={resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a result with a concept display', () => {
    const resultData = {
      ...DefaultResult,
      conceptDisplay: 'Temperature Oral',
    };
    const result = shallow(<ClinicalResult resultData={resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a result with a date time display', () => {
    const resultData = {
      ...DefaultResult,
      datetimeDisplay: 'Nov 23, 2019 13:31:31',
    };
    const result = shallow(<ClinicalResult resultData={resultData} />);
    expect(result).toMatchSnapshot();
  });
});

describe('ClinicalResultBloodPressure', () => {
  it('should render a default ClinicalResultBloodPressure', () => {
    const result = shallow(<ClinicalResultBloodPressure resultData={DefaultBloodPressureResult} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResultBloodPressure', () => {
    const result = shallow(<ClinicalResultBloodPressure resultData={DefaultBloodPressureResult} isTruncated />);
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResultBloodPressure with icons moved', () => {
    const resultData = {
      id: '111',
      systolic: {
        ...DefaultSystolicResult,
        isModified: true,
      },
      diastolic: {
        ...DefaultDiastolicResult,
        isModified: true,
      },
    };
    const result = shallow(<ClinicalResult resultData={resultData} isTruncated />);
    expect(result).toMatchSnapshot();
  });

  describe('- Systolic -', () => {
    it('should pass certain props to Observation ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          interpretation: 'CRITICAL',
          type: 'ALPHA',
          status: 'registered',
          performedDateTime: new Date('December 17, 1995 03:24:00'),
          updateDateTime: new Date('October 21, 1995 04:24:00'),
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} hideUnit />);
      expect(result).toMatchSnapshot();
    });

    it('should render a modified result ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          isModified: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a commented result ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          hasComment: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render an unverified result ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          isUnverified: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a result with unit hidden', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          hideUnit: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a concept display', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          conceptDisplay: 'Temperature Oral',
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a date time display', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          datetimeDisplay: 'Nov 23, 2019 13:31:31',
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });
  });

  describe('- Diastolic -', () => {
    it('should pass certain props to Observation ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          interpretation: 'CRITICAL',
          type: 'ALPHA',
          status: 'registered',
          performedDateTime: new Date('December 17, 1995 03:24:00'),
          updateDateTime: new Date('October 21, 1995 04:24:00'),
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} hideUnit />);
      expect(result).toMatchSnapshot();
    });

    it('should render a modified result ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          isModified: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a commented result ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          hasComment: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render an unverified result ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          isUnverified: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a result with unit hidden', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          hideUnit: true,
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a concept display', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          conceptDisplay: 'Temperature Oral',
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a date time display', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          datetimeDisplay: 'Nov 23, 2019 13:31:31',
        },
      };
      const result = shallow(<ClinicalResultBloodPressure resultData={resultData} />);
      expect(result).toMatchSnapshot();
    });
  });
});
