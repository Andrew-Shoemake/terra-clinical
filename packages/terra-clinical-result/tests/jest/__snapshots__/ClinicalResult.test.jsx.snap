// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicalResult should pass certain props to Observation  1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={true}
        interpretation="CRITICAL"
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResult should render a commented result  1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
      <IconComment
        className="icon-comment"
        data-name="Layer 1"
        isBidi={true}
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResult should render a default ClinicalResult 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResult should render a modified result  1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
      <IconModified
        className="icon-modified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResult should render a result with a concept display 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Temperature Oral
  </div>
</div>
`;

exports[`ClinicalResult should render a result with a date time display 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
  </div>
  <div
    className="datetime-display"
  >
    Nov 23, 2019 13:31:31
  </div>
</div>
`;

exports[`ClinicalResult should render a result with unit hidden 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResult should render a truncated ClinicalResult 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResult should render a truncated ClinicalResult with icons moved 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
    </div>
    <IconModified
      className="icon-modified"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
</div>
`;

exports[`ClinicalResult should render an unverified result  1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111"
        hideUnit={false}
        isTruncated={false}
        result={
          Object {
            "unit": "mL",
            "value": "12345.678",
          }
        }
      />
      <IconDiamond
        className="icon-unverified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should pass certain props to Observation  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={true}
        interpretation="CRITICAL"
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a commented result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
      <IconComment
        className="icon-comment"
        data-name="Layer 1"
        isBidi={true}
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a modified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
      <IconModified
        className="icon-modified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a result with a concept display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Temperature Oral
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a result with a date time display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Nov 23, 2019 13:31:31
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a result with unit hidden 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render an unverified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ResultError
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.2"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
      <IconDiamond
        className="icon-unverified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should pass certain props to Observation  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={true}
        interpretation="CRITICAL"
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a commented result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
      <IconComment
        className="icon-comment"
        data-name="Layer 1"
        isBidi={true}
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a modified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
      <IconModified
        className="icon-modified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a result with a concept display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Temperature Oral
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a result with a date time display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Nov 23, 2019 13:31:31
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a result with unit hidden 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render an unverified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <ResultError
        key="Error-Diastolic-111"
      />
      <IconDiamond
        className="icon-unverified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure should render a default ClinicalResultBloodPressure 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={true}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure should render a truncated ClinicalResultBloodPressure 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={true}
        isTruncated={false}
        key="Observation-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="ObservationSeperator-111.1"
      >
        /
      </span>
      <Observation
        eventId="111.2"
        hideUnit={false}
        isTruncated={false}
        key="Observation-111.2"
        result={
          Object {
            "unit": "mmHg",
            "value": "80",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure should render a truncated ClinicalResultBloodPressure with icons moved 1`] = `
<div
  className="clinical-result"
>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <Observation
        hideUnit={false}
        isTruncated={false}
        result={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`should render a default ResultTimeHeaderCell component 1`] = `
<div
  className="clinical-result-time-header-cell padding-compact"
>
  <div
    className="date"
  >
    Dec 21, 2010
  </div>
  <div
    className="time"
  >
    23:59
  </div>
</div>
`;

exports[`should render a default component 1`] = `
<div
  className="clinical-result-name-header-cell padding-compact"
>
  <div
    className="name"
  >
    Blood Pressure
  </div>
  <div
    className="unit"
  >
    mmHg
  </div>
</div>
`;
